File: 백준/Silver/1966. 프린터 큐/프린터 큐.py
import sys
input = sys.stdin.readline

test = int(input())

for _ in range(test):
    n, m = map(int, input().split())
    queue_list = list(map(int, input().split()))
    queue = []
    for i in range(n):
        queue.append((queue_list[i], i))
    
    sort_queue = []
    while len(sort_queue) != n:
        priority, index = queue.pop(0)
        # print(priority, index, queue)
    
        if priority < max(queue_list):
            queue.append((priority, index))
        else:
            sort_queue.append(index)
            queue_list.remove(priority)
        
    print(sort_queue.index(m)+1)

File: 백준/Silver/1966. 프린터 큐/프린터 큐.js
const input = require("fs")
  .readFileSync(
    process.platform === "linux" ? "/dev/stdin" : __dirname + "/input.txt"
  )
  .toString()
  .split("\n");

const test_case = Number(input.shift());

for (let i = 0; i < test_case; i++) {
  const [n, m] = input.shift().split(" ").map(Number);

  const docs = input
    .shift()
    .split(" ")
    .map((priority, index) => ({
      priority: Number(priority),
      index,
    }));

  let cnt = 1;
  let max_priority = Math.max(...docs.map((doc) => doc.priority));

  while (true) {
    const { priority, index } = docs.shift();

    if (priority !== max_priority) {
      docs.push({ priority, index });
    } else {
      if (index === m) {
        console.log(cnt);
        break;
      }

      max_priority = Math.max(...docs.map((doc) => doc.priority));
      cnt++;
    }
  }
}
